test/test_database_integrity/test_database_integrity

```diff
@@ -0,0 +1,372 @@
+import os
```

Comment:
@hugwuoke I'm going to review and merge this some time soon. If I merge it before you pick up on it again, we can continue any work needed in a new PR.

Only thing I noticed here at a glance is that `test_database_integrity` needs the file extension `.py` at the end!